lib/jobs/ScheduleBackupJob.js

```diff
@@ -132,6 +136,21 @@ class ScheduleBackupJob extends BaseJob {
               return deleteResponse;
             }
           });
+      })
+      .then(deleteResponse => {
```

Comment:
why are we doing this here instead of doing it initially itself i.e [here](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/e627fe9890a4134c88b9ab6f4385cffabdf23d07/lib/jobs/ScheduleBackupJob.js#L101)  ?

Also why are we hard-coding iaas name here ? can this based on presence on some config or if we have `alternate_retention_period` ?

Reply:
Yes, @jagadish-kb also had the same review comment. I have made the changes based on that. We will be checking for the alternate retention period (e.g. azure_cost_reduction_retention_period) and delete in the same job. The changes are being tested and will be pushed.